packages/bpk-component-card-list/src/BpkExpand/BpkExpand.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BUTTON_TYPES, BpkButtonV2 } from '../../../bpk-component-button';
+import {
+  withButtonAlignment,
+  withRtlSupport,
+} from '../../../bpk-component-icon';
+import ChevronDown from '../../../bpk-component-icon/sm/chevron-down';
+import ChevronUp from '../../../bpk-component-icon/sm/chevron-up';
+
+import type { ExpandProps } from '../common-types';
+
+const AlignedChevronDownIcon = withButtonAlignment(withRtlSupport(ChevronDown));
+const AlignedChevronUpIcon = withButtonAlignment(withRtlSupport(ChevronUp));
+
+const BpkExpand = ({
+  children,
+  collapsed,
+  hideContent,
+  setCollapsed,
+  showContent,
+}: ExpandProps) => {
+  const buttonIcon = collapsed ? (
+    <AlignedChevronDownIcon />
+  ) : (
+    <AlignedChevronUpIcon />
+  );
+  const buttonOnClick = () => {
+    if (collapsed) {
+      showContent();
+      setCollapsed(false);
```

Comment:
I think it's better not to directly change the state here and instead pass a callback from the parent for the onClick functionality. In fact, I think `showContent` handles that already